### ISSUE TITLE
Docs: Update privacy policy, system requirements and FAQ to better inform about external services

### ISSUE THREAD

**Issue by WhyNotHugo** [photoprism/photoprism#2507]
#### 1. What is not working as documented?

Photoprism silently sends the location data for ALL PHOTOS to https://places.photoprism.app.

I did a quick try of photoprism on a container with no network access (but based mostly on `https//dl.photoprism.app/docker/docker-compose.yml`). As soon as I started indexing photos, I saw a massive amount of failed network requests, each sending the location of an individual photo.

Here's the output of one example https://places.photoprism.app/v1/location/47c609a52074:

```json
{
  "id": "s2:47c609a52074",
  "name": "Steiger 3",
  "street": "Oosterdok",
  "postcode": "1011VZ",
  "category": "",
  "timezone": "Europe/Amsterdam",
  "lat": 52.374054,
  "lng": 4.910662,
  "place": {
    "id": "nl:eOoRn3FCZyVe",
    "label": "Amsterdam, Noord-Holland, Netherlands",
    "district": "Centrum",
    "city": "Amsterdam",
    "state": "Noord-Holland",
    "country": "nl",
    "keywords": ""
  },
  "events": [],
  "licence": "Data © PhotoPrism"
}
```

Given that photoprism seems to attempt this for every photo, the information leaked here is the exact location of every photo I've ever taken.

However, the [privacy policy](https://photoprism.app/privacy) is very clear in stating that this is not a thing:

> (a) Self-hosting is the easiest way to stay in control and protect your privacy. Data that never leaves your private network cannot be collected by anyone, including us.

#### 2. How can we reproduce it?

1. Run the sample in `https//dl.photoprism.app/docker/docker-compose.yml`
2. Inspect network traffic

OR

1. Run the sample with this diff applied (that puts photoprism in a network with no internet access):

```diff
--- orig	2022-07-08 13:13:01.867412647 +0200
+++ docker-compose.yml	2022-07-08 12:58:16.084856426 +0200
@@ -105,10 +105,12 @@
     ## Storage Folders: "~" is a shortcut for your home directory, "." for the current directory
     volumes:
       # "/host/folder:/photoprism/folder"                # Example
-      - "~/Pictures:/photoprism/originals"               # Original media files (DO NOT REMOVE)
+      - "~/photos:/photoprism/originals:ro"               # Original media files (DO NOT REMOVE)
       # - "/example/family:/photoprism/originals/family" # *Additional* media folders can be mounted like this
       # - "~/Import:/photoprism/import"                  # *Optional* base folder from which files can be imported to originals
       - "./storage:/photoprism/storage"                  # *Writable* storage folder for cache, database, and sidecar files (DO NOT REMOVE)
+    networks:
+      - internal
 
   ## Database Server (recommended)
   ## see https://docs.photoprism.app/getting-started/faq/#should-i-use-sqlite-mariadb-or-mysql
@@ -131,6 +133,17 @@
       MARIADB_USER: "photoprism"
       MARIADB_PASSWORD: "insecure"
       MARIADB_ROOT_PASSWORD: "insecure"
+    networks:
+      - internal
+
+  caddy:
+    image: caddy
+    ports:
+      - "9000:9000"
+    command: "caddy reverse-proxy --from localhost:9000 --to photoprism:2342"
+    networks:
+      - internal
+      - dmz
 
   ## Watchtower upgrades services automatically (optional)
   ## see https://docs.photoprism.app/getting-started/updates/#watchtower
@@ -144,3 +157,9 @@
   #   volumes:
   #     - "/var/run/docker.sock:/var/run/docker.sock"
   #     - "~/.docker/config.json:/config.json" # optional, for authentication if you have a Docker Hub account
+
+networks:
+  dmz:
+    internal: false
+  internal:
+    internal: true
```

#### 3. What behaviour do you expect?

Either of the following would be reasonable:

1. Ask for explicit consent before uploading the location data of an entire photo library to a remote/online service.
2. Resolve locations offline/locally; for example, downloading a large (somewhat curated) dump from OSM.

Additionally, the privacy policy needs to be updated: it clearly states that no data is uploaded, while clearly very sensitive data is being uploaded.

#### 4. What could be the cause of your problem?

An oversight in development -- this seems to work the way it does by design.

#### 5. Can you provide us with example files for testing, error logs, or screenshots?

Nothing special to reproduce this; reproducible out-of-the-box with no special data or configuration. 

#### 6. Which software versions do you use?

(a) PhotoPrism Architecture & Build Number: amd64, docker image b4f1c6a5b5385e23d27f19cad2f2ecc59e9015bdd398b2059432b95ac793b545

(b) Database Type & Version: MariaDB

(c) Operating System Types & Versions: Linux,

(d) Browser Types & Versions: Irrelevant

(e) Ad Blockers, Browser Plugins, and/or Firewall Software? See above

#### 7. On what kind of device is PhotoPrism installed?

Laptop, all details are unimportant here.

#### 8. Do you use a Reverse Proxy, Firewall, VPN, or CDN?

See above.

**Comment by lastzero** [comment]
This is written on our feature page, in our compliance FAQ, the privacy policy, and the developer guide. RTFM.

**Comment by lastzero** [comment]
Why don't you submit a PR with a consent dialog that meets your needs?

**Comment by WhyNotHugo** [comment]
> This is written on our feature page, in our compliance FAQ, the privacy policy

The privacy policy (as I quoted above) states the exact opposite (emphasis mine):

> (a) Self-hosting is the easiest way to stay in control and protect your privacy. **Data that never leaves your private network cannot be collected by anyone, including us**.

**Comment by WhyNotHugo** [comment]
Looking at the [FAQ](https://docs.photoprism.app/getting-started/faq/) (I read the questions, not the answers), nothing seems to be related. I searched (e.g.: Ctrl+F) for `location`, and the only mention of this word is unrelated to the issue at hand.

**Comment by WhyNotHugo** [comment]
The [feature page](https://docs.photoprism.app/) also does not seem to mention this.

Actually, all these pages focus a lot on privacy; none of them mention "the location data for all your photos is submitted over the internet to a remote server".

**Comment by lastzero** [comment]
**Does your software depend on external services or is it completely self-contained?**

As explained in detail in our Privacy Policy, reverse geocoding and interactive world maps depend on retrieving the necessary data from external systems supplied by us and MapTiler AG, headquartered in Switzerland. Both services are provided with a very high level of privacy and confidentiality.

Your use of these services is fully covered by us. So there are no usage-based fees, unlike with other vendors who may charge additional fees and may also not allow you to use caching, e.g. Google Maps.

Should you still wish to operate one or both of these services on your own premises, we can set up such a fully autonomous solution for you, provided you are prepared to cover the initial setup costs as well as ongoing maintenance fees for content licenses and updates.

**Comment by lastzero** [comment]
### 7. Reverse Geocoding API

If enabled, our privacy-preserving reverse geocoding service provides global location information to enrich your photo and video metadata with details such as country, state, city, and category:

(a) API requests are not logged permanently.

(b) Location data is cached so that it can be reused for subsequent requests from you or other users.

(c) Our API approximates the coordinates and encodes them with a fuzzy S2 cell ID that does not include the house number or any other data identifying a specific residential address, except possibly in very sparsely populated areas of the world. Even then, we cannot trace the request back to a person, picture or point in time.

(d) We may store your server's IP address and other HTTP request headers for a limited time to perform authorization checks, prevent abuse, and implement rate limits. Since the traffic is encrypted, no one intercepting the server-to-server communication can see the exact request and response; only the fact that you exchanged data with our backend.

## 8. Interactive World Maps

(a) When you use the interactive world maps integrated in our apps and services (with the exception of the low-detail offline map), MapTiler AG, based in Zug, Switzerland, stores your IP address for a limited period of time as required to perform security checks; the documented maximum is 20 minutes. After that, it is automatically discarded.

(b) This is necessary to detect and block malicious activity on their infrastructure, as well as for accounting purposes. Visit maptiler.com/privacy-policy to learn more.

**Comment by lastzero** [comment]
"Our privacy-preserving backend services provide worldwide location information to enrich your pictures with details such as state, city, and category."

Backend, services, provide are the keywords here.

**Comment by WhyNotHugo** [comment]
I see. I seem to have missed that when reading through the details.

Honestly, I saw a couple of mentions of you guys, and was curious whether it worked offline. I looked up the face recognition part, and saw no mention of photos or metadata being uploaded anywhere. That seemed nice, so I read the privacy policy and came across this:

> (a) Self-hosting is the easiest way to stay in control and protect your privacy. Data that never leaves your private network cannot be collected by anyone, including us.

This seems to hint that data does not leave my network. I think this needs to be removed, the way it's written is very misleading. It gives the impression that, by self hosting, my data would not leave my network. A rather sensitive subset of the data does leave my private network.

> Why don't you submit a PR with a consent dialog that meets your needs?

Clearly, I can figure out that this leak is going on and it has been dropped anyway (given that the service runs with no internet access). I worry that most users will not be aware of this going on under the hood, and transmitting sensitive information to a remote service provider requires user consent, IMHO.

The sum of the location history of a person's photo album can be an extremely detailed record of places a person has been over years. This is transmitted together with a person's [IP address](https://ec.europa.eu/info/law/law-topic/data-protection/reform/what-personal-data_en), considered PII in the EU.

Of course, I understand that you are not service providers in this case, and source code is made available with no warranties. I'm mostly pointing this out as as FYI; as a project which identifies itself with privacy awareness, I believe this should be of interest to you.

> Backend, services, provide are the keywords here.

It would never have occurred to me to search under those keywords. The backend runs on my local machine (photoprism itself). I've no idea why you'd call a remote location resolving service the backend. If places.photoprism.app is considered the backend, then [this docker-compose file](https//dl.photoprism.app/docker/docker-compose.yml) isn't really self-hosting, it's only running a frontend/client, right?

**Comment by lastzero** [comment]
If a confirm dialog is important to you, you are welcome to contribute one. In practice, the requests are completely anonymous. We have no information other than that photos were taken on planet earth. Keep in mind the huge number of requests and users. If you work for a government and have higher security standards, we can provide that for a fee.

**Comment by lastzero** [comment]
Data THAT never leaves. For some features, you still need data, but we don't UPLOAD FILES FOR THIS. We did not write you get the satellite and satellite maps on top of the free software. Read carefully and until the end. This clearly would have helped your understanding.

**Comment by WhyNotHugo** [comment]
So when you put these two sentences together:

> Self-hosting is the easiest way to stay in control and protect your privacy. Data that never leaves your private network cannot be collected by anyone, including us.´

It strongly implies that because one self hosts, the data won't leave the local network. It's not worded literally, but I think it's quite natural that someone might expect this.

> If a confirm dialog is important to you, you are welcome to contribute one. In practice, the requests are completely anonymous. We have no information other than that photos were taken on planet earth. Keep in mind the huge number of requests and users. If you work for a government and have higher security standards, we can provide that for a fee.

Honestly, my need are met; I can figure these details out myself. I just wish you'd be more transparent with _other_ users, making it clear to them what information is being transmitted to someone else over the internet. "self hosted" seldom implies "all location metadata is sent out". I wonder if most users have any idea of what's actually going on. I'm sure a privacy centric crowd would care about this. It definitely sounds like the kind of setting someone privacy-ware would expect disabled by default.

Then again, we live in a world where companies like Facebook proudly lie "we care about privacy", so I guess I should be used to see this label put on products that are not clear about sharing sensitive data in the least.

**Comment by lastzero** [comment]
You compare us with Facebook? 🤡

**Comment by WhyNotHugo** [comment]
I did not. I won't engage in further discussion here. It seems that you're mostly claiming to care about privacy more the marketing value, but don't actually care about this at all.

I do hope you're reconsider asking users for their consent before extracting all their location metadata from their photos and sending them to remote servers. Especially while giving the impression that the product is self-hosted and data is retained locally. What you're doing might be legal, but it's hardly ethical.

**Comment by lastzero** [comment]
I guess your point is that we should start charging for this service so that users like you become more aware of the value they receive. Is that correct?

**Comment by heikomat** [comment]
It seems this discussion here got a little heated and emotional. Please allow me to try to make it more constructive by summarizing the (from my point of view) valid points from both sides.

First of @WhyNotHugo I can somewhat see why you might think that the way things are formulated **might be** a little misleading to some people. This doesn't mean that "everything gets leaked".

I would consider myself pretty privacy concious, and when i first interacted with photoprism, i read the same passages, but it was clear to me: When i want to be able to search for cities and places, i'd have to send these location-data (and nothing else) to that outside-service. But that was fine for me, because:
- I wanted that functionality
- Photoprism is open source, so i can guarantee that only that information is sent, and nothing else
- I trust places.photoprism.app more with some location info than i do google with ALL the photos and metadata.

That being said:
@lastzero That outside service it is mentioned multiple times and if someone reads the privacy policy and FAQ just a little more careful, that someone should know the implications. I'm with you here. **BUT** A lot of people don't. I can easily imagine people reading: "open source", "self hosted" and directly set things up thinking everything is local.

Maybe we should at least consider one of these options (doesn't mean i want them implemented, just that we might consider them):
1. having the location feature off by default, so that everything is actually local. When the user goes to enable that feature in the UI, he/she might get a popup that informs about the usage of that outside service
2. Having the feature off by default, but on the very first visit (when the password isn't even updated yet), show a popup or page or something where he/she is asked, if he/she wants to enable that functionality (again, informing about that outside service)
3. Allow the user to self host whatever runs on `places.photoprism.app` (would be nice, but don't know how feasable/possible that is)

That second option would for example fit nicely into an onboarding wizard! (https://github.com/photoprism/photoprism/issues/652)

-----------------------

TL;DR:
- @WhyNotHugo has some valid points, but saying everything gets leaked is neither fair nor correct
- @lastzero has done enough to inform the user, and can't be blamed. That doesn't mean that the user experience can't be improved!

**Comment by lastzero** [comment]
Most users would absolutely hate having to reindex all pictures because it was turned off. The correct way is to implement a confirm / consent dialog which I currently don't have time for. One of the reasons is that people use bug reports for discussions. I've asked many times to please contribute such a dialog, but no. He probably thinks we have a marketing department that could do it. So mistaken.

**Comment by lastzero** [comment]
We don't benefit from offering this service. It's not like we collect data to sell it. Others charge for it and if our users want us to develop an army of dialogs and legal documents, we have to charge them too. What about saying THANK YOU for doing all this!?

**Comment by graciousgrey** [comment]
I would also not turn it off by default but I like the idea to include it in the onboarding wizard when we find time to implement it :)

**Comment by FlorianCassayre** [comment]
+1 for a wizard upon setup.

**Comment by lastzero** [comment]
Despite the unnecessarily harsh comments above, and despite the fact that we're on vacation right now, I've just updated our privacy policy, system requirements, and FAQ to better inform about external services:

- https://photoprism.app/privacy#not-collected
- https://docs.photoprism.app/getting-started/faq/#does-photoprism-depend-on-any-external-services
- https://docs.photoprism.app/user-guide/faq/
- https://docs.photoprism.app/getting-started/#external-services

These pages now contain explicit notices and highly visible links to the relevant sections in our Compliance FAQ and Privacy Policy (which were already available):

- https://photoprism.app/kb/compliance-faq#privacy
- https://photoprism.app/privacy#section-7

**We hope this helps you make an informed decision, because it is absolutely not our intention to collect information beyond what is technically necessary:**

(a) to provide the services (most) users expect when they use PhotoPrism and 
(b) to prevent abuse, e.g. to protect our infrastructure and your data as far as it is stored with us

**Comment by lastzero** [comment]
@FlorianCassayre Would you like to work on a wizard? We appreciate any contribution.

@WhyNotHugo We've added a big warning right on the setup page at https://docs.photoprism.app/getting-started/#maps-places and also inform all users who don't support us that they are incurring costs on our side by using our services, which will hopefully make them a bit more sensitive when they complain.
